EIPS/eip-4512.md

```diff
@@ -0,0 +1,168 @@
+status: Draft
+type: Standards Track
+author: KKimos (@KKimos), KKimos <bol@zju.edu.cn>
+category: ERC
+created: 2021-12-01
+requires: 165, 721
+---
+
+## Simple Summary
+Only add a user role, change the way NFT sharing.
+
+## Abstract
+
+This standard adds a new right of Non-Fungible Token that the right to use. Through this standard, you can achieve :
+
+- Separation of the right to use and ownership of Non-Fungible Token
+- Non-secured lease Non-Fungible Token
+- You can continue to use it after you mortgage the Non-Fungible Token
+- Metaverse sharing economy
+
+It is precisely because of the separation of ownership and use right that the utilization rate of assets can be greater. You must distinguish between the rights of the user and the owner.
+
+## Motivation
+
+Uber, airbnb, and WeWork promote the development of the sharing economy. In the blockchain , the ERC721-based leasing system has encountered great bottlenecks. For example, excess assets need to be mortgaged for leasing, The increase in NFT prices will lead to defaults ,You cannot continue to use it during NFT in mortgages. As a result, a new NFT standard is proposed, which is fully compatible with ERC721. Separate the right to use and own the NFT.
+
+Different from 2615, we only added a right to use, so that we can get all the functions of 2615, and reduce many unnecessary operations in 2615. The explanation is as follows:
+
+- The owner should not use some of the ownership rights when mortgage NFT, such as transfer, AXS-like breeding or weapon upgrade, because it is likely to change the status of the original NFT, so when the user mortgages the NFT, the ownership must be mortgaged in the contract middle.
+- In the case of mortgage or transaction, the owner has the right to continue to use it before returning the ransom or selling it
+
+## Specification
+
+This standard adds the user role, and users can transfer their own usage rights.
+
+### ERC-4512 Interface
+
+```solidity
+event TransferUser(address from,address to,uint256 tokenId);
+event ApprovalUser(address indexed user, address indexed approved, uint256 indexed tokenId);
+function balanceOfUser(address user) external view returns (uint256 balance);
+function userOf(uint256 tokenId) external view returns (address user);  
+function safeTransferUserFrom(
+    address from,
+    address to,
+    uint256 tokenId
+) external;
+function safeTransferUserFrom(
+    address from,
+    address to,
+    uint256 tokenId,
+    bytes calldata data
+) external;
+function safeTransferAllFrom(
+     address from,
+     address to,
+     uint256 tokenId
+) external;    
+function safeTransferAllFrom(
+     address from,
+     address to,
+     uint256 tokenId,
+     bytes calldata data
+) external;
+function approveUser(address to, uint256 tokenId) external;
+function getApprovedUser(uint256 tokenId) external view returns (address operator);
```

Comment:
personally i do feel that including doc strings such as natspec may be helping in getting the reader up to speed